Confusion with weekly byweekday and toText()

``` javascript
var RRule = require('../')
var rrulestr = RRule.rrulestr
var assert = require('assert')

describe.only('Is this broken', function () {

  it('knows the days of the week', function() {
    var r = RRule.rrulestr('FREQ=WEEKLY;BYDAY=TU');
    assert.equal(r.toString(), 'FREQ=WEEKLY;BYDAY=TU');
    assert.equal(r.toText(), 'every week on Tuesday');
  });

  it('does not know the days of the week', function() {
    var r = new RRule({
      freq: RRule.WEEKLY,
      byweekday: 1
    });
    assert.equal(r.toString(),'FREQ=WEEKLY;BYDAY=TU');
    assert.equal(r.toText(),'every week on Tuesday');
  });

});
```

generates the output

```
  Is this broken
    ✓ knows the days of the week
    1) does not know the days of the week


  1 passing (11ms)
  1 failing

  1) Is this broken does not know the days of the week:

      AssertionError: 'every week on Monday' == 'every week on Tuesday'
      + expected - actual

      -every week on Monday
      +every week on Tuesday
```

which is at best confusing.

But I can't see a simple and elegant way out of it.  Am I missing something?

--- a/lib/nlp.js
+++ b/lib/nlp.js
@@ -396,7 +396,7 @@
       },
 
       weekdaytext: function (wday) {
-        var weekday = typeof wday === 'number' ? wday : wday.getJsWeekday()
+        var weekday = typeof wday === 'number' ? (wday + 1) % 7 : wday.getJsWeekday()
         return (wday.n ? this.nth(wday.n) + ' ' : '') +
           this.language.dayNames[weekday]
       },

